src/overlays/actors/ovl_En_Ru2/z_en_ru2.c

```diff
@@ -575,65 +727,89 @@ void func_80AF3564(EnRu2* this, PlayState* play) {
+    EnRu2_UpdateEyes(this);
+    EnRu2_FadeInCredits(this);
+    EnRu2_CheckVisibleInCredits(this);
 }
 
-void func_80AF366C(EnRu2* this, PlayState* play) {
-    func_80AF2744(this, play);
+/**
+ * Action 12
+ */
+void EnRu2_CreditsVisible(EnRu2* this, PlayState* play) {
+    EnRu2_UpdateBgCheckInfo(this, play);
     EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF3564(this, play);
+    EnRu2_UpdateEyes(this);
+    EnRu2_NextCreditsAction(this, play);
 }
 
-void func_80AF36AC(EnRu2* this, PlayState* play) {
-    s32 something;
+/**
+ * Action 13
+ */
+void EnRu2_CreditsTurnHeadDownLeft(EnRu2* this, PlayState* play) {
+    s32 animDone;
 
-    func_80AF2744(this, play);
-    something = EnRu2_UpdateSkelAnime(this);
-    func_80AF2608(this);
-    func_80AF3530(this, something);
+    EnRu2_UpdateBgCheckInfo(this, play);
+    animDone = EnRu2_UpdateSkelAnime(this);
+    EnRu2_UpdateEyes(this);
+    EnRu2_LookingDownLeft(this, animDone);
 }
 
-void func_80AF36EC(EnRu2* this, PlayState* play) {
-    Flags_SetSwitch(play, func_80AF2690(this));
+/**
+ * Sets the switch indicating that Link met Ruto in the Water Temple.
+ */
+void EnRu2_MarkEncounterOccurred(EnRu2* this, PlayState* play) {
+    Flags_SetSwitch(play, EnRu2_GetSwitchFlag(this));
 }
 
-s32 func_80AF3718(EnRu2* this, PlayState* play) {
-    return Flags_GetSwitch(play, func_80AF2690(this));
+/**
+ * Returns whether Link met Ruto in the Water Temple.
+ */
+s32 EnRu2_EncounterOccurred(EnRu2* this, PlayState* play) {
+    return Flags_GetSwitch(play, EnRu2_GetSwitchFlag(this));
 }
 
-void func_80AF3744(EnRu2* this, PlayState* play) {
-    if (func_80AF3718(this, play)) {
+/**
+ * Initializes Ruto's actor in the Water Temple, or destroys it if the encounter already happened.
+ */
+void EnRu2_InitWaterTempleEncounter(EnRu2* this, PlayState* play) {
+    if (EnRu2_EncounterOccurred(this, play)) {
         Actor_Kill(&this->actor);
     } else {
-        func_80AF28E8(this, &gAdultRutoIdleAnim, 0, 0.0f, 0);
-        this->action = 14;
-        this->drawConfig = 1;
+        EnRu2_AnimationChange(this, &gAdultRutoIdleAnim, 0, 0.0f, 0);
+        this->action = ENRU2_WATER_TEMPLE_ENCOUNTER_RANGE_CHECK;
+        this->drawConfig = ENRU2_DRAW_OPA;
     }
 }
 
-void func_80AF37AC(void) {
+void EnRu2_PlayFanfare(void) {
     Audio_PlayFanfare(NA_BGM_APPEAR);
 }
 
-void func_80AF37CC(EnRu2* this) {
+/**
+ * Accelerates Ruto's actor upwards as she swims.
+ */
+void EnRu2_AccelerateUp(EnRu2* this) {
     f32 funcFloat;
 
-    this->unk_2C0++;
-    funcFloat = Environment_LerpWeightAccelDecel((kREG(2) + 0x96) & 0xFFFF, 0, this->unk_2C0, 8, 0);
+    this->swimmingUpFrame++;
+    funcFloat = Environment_LerpWeightAccelDecel((kREG(2) + 0x96) & 0xFFFF, 0, this->swimmingUpFrame, 8, 0);
     this->actor.world.pos.y = this->actor.home.pos.y + (300.0f * funcFloat);
 }
 
-s32 func_80AF383C(EnRu2* this, PlayState* play) {
+s32 EnRu2_IsPlayerInRange(EnRu2* this, PlayState* play) {
```

Comment:
In range for what? it has a very specific position check. 
Seems to be for starting the encounter, so `EnRu2_IsPlayerInRangeForEncounter`